src/binarytools/BinaryReader.cpp

```diff
@@ -192,6 +192,21 @@ std::string Ship::BinaryReader::ReadString() {
     return res;
 }
 
+std::string Ship::BinaryReader::ReadCString() {
+    std::string res;
+
+    unsigned char c = 0;
+    do {
+        if (mStream->GetBaseAddress() >= mStream->GetLength())
```

Comment:
Why not put this in the while? I'm OK with either, just curious on the preference.

Reply:
I usually don't use do-while loops but in this case I felt it made sense. It needs to read at least one character before it knows whether or not to stop (checking for the terminator), a do-while guarantees that. A pure-while would have to be written a bit differently.